runtime.go

```diff
@@ -591,7 +625,79 @@ func getWorkingDir(basedir string, codeuri string, checkWorkingDirExist bool) (s
 
 	// Windows uses \ as the path delimiter, but Docker requires / as the path delimiter.
 	dir = filepath.ToSlash(dir)
-
 	return dir, nil
 
 }
+
+// decompressArchive unzips a ZIP archive to a temporary directory and returns
+// the temporary directory name, or an error
+func decompressArchive(src string) (string, error) {
+
+	// Create a temporary directory just for this decompression (dirname: OS tmp directory + unix timestamp))
+	tmpdir := os.TempDir()
+
+	// By default on OSX, os.TempDir() returns a directory in /var/folders/.
+	// This sits outside the default Docker Shared Files directories, however
```

Comment:
We need to document the folders where tmp dir will be created. Customers might need to explicitly whitelist this in Docker File Sharing Preferences